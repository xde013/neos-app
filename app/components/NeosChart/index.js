/**
 *
 * NeosChart
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Chart } from 'react-google-charts';
import LoadingIndicator from '../LoadingIndicator';

function NeosChart({ neos }) {
  return (
    <Chart
      width="100%"
      height="65vh"
      chartType="BarChart"
      loader={<LoadingIndicator />}
      data={[
        ['Neo', 'Min Estimated Diameter (km)', 'Max Estimated Diameter (km)'],
        ...neos,
      ]}
      options={{
        title: 'NEOs diameters',
        hAxis: {
          title: 'NEOs diameters',
          gridlines: {
            count: 10,
          },
          viewWindow: { min: 0 },
        },
        vAxis: {
          viewWindowMode: 'pretty',
          title: 'NEOs',
          slantedText: true,
          maxTextLines: 1,
          direction: 1,
          textPosition: 'out',
          textStyle: {
            fontSize: 12,
            bold: true,
          },
        },
        legend: { position: 'top' },
      }}
      // For tests
      rootProps={{ 'data-testid': '1' }}
    />
  );
}

NeosChart.propTypes = {
  neos: PropTypes.array.isRequired,
};

export default NeosChart;
