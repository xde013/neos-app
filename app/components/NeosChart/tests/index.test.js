/**
 *
 * Tests for NeosChart
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import NeosChart from '../index';

describe('<NeosChart />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<NeosChart neos={[]} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<NeosChart neos={[]} />);
    expect(firstChild).toMatchSnapshot();
  });
});
