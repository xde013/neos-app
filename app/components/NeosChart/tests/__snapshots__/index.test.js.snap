// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NeosChart /> Should render and match the snapshot 1`] = `
<div
  class="LoadingIndicator__Wrapper-c3cbkl-0 cnqvsL"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
